### PR TITLE
More kernel tweaks

### PR DESCRIPTION
**Pull request by fingolfin**
With this, the segfault in the test suite seems to be gone for me.

There are more things that could and should be cleaned up in the kernel and library code of this package. I'd submit some thoughts and suggestions on this to your issue tracker, but you may want to resolve https://github.com/kamalsaleh/CddInterface/issues/9 first.

### CODE REVIEW

File: src/CddInterface.c [hunk 1]
```diff
@@ -38,8 +27,8 @@ extern void dd_SetLinearity(dd_MatrixPtr, char *);
 // }
 
 // The following conversion has been taken from
-// https://github.com/gap-packages/NormalizInterface/src
-// Thanks to Sebastian Gutsche
+// https://github.com/gap-packages/NormalizInterface
+// Thanks to Max Horn
```
Comment (fingolfin) [inline src/CddInterface.c:31]: Just to be clear, I am also fine with not getting thanked here, but if there are thanks, then to the author, I'd say ;-).

File: src/CddInterface.c [hunk 2]
```diff
@@ -38,8 +27,8 @@ extern void dd_SetLinearity(dd_MatrixPtr, char *);
...
 static Obj MPZ_TO_GAPOBJ( const mpz_t x)
 {
```
Comment (fingolfin) [inline src/CddInterface.c:33]: If you are willing to require GAP >= 4.10, you could also replace the implementation of this function by a single line:
```C
  return MakeObjInt((const UInt *)x->_mp_d, x->_mp_size);
```

File: src/CddInterface.c [hunk 3]
```diff
@@ -278,23 +257,20 @@ static long int *ddG_InteriorPoint(dd_MatrixPtr M)
...
     {
-
-      mpq_get_num(u, lps->sol[j]);
-      mpq_get_den(v, lps->sol[j]);
-      z1 = mpz_get_si(u);
-      z2 = mpz_get_si(v);
-
-      result[2 * (j - 1) + 1] = z1;
-      result[2 * (j - 1) + 2] = z2;
+      ASS_LIST(result, j + 1, MPQ_TO_GAPOBJ(lps->sol[j]));
```
Comment (fingolfin) [inline src/CddInterface.c:249]: Not only is this much simpler, and avoids passing data through a global variable (always bad style), it also works correctly with data that does not fit into a GAP intobj.

File: gap/tools.gi [hunk 4]
```diff
@@ -109,10 +109,12 @@ InstallMethod( IsCompatiblePolyhedronList,
     
     fi;
     
-    if not ( Length( list[ 7 ]  ) = 2 * list[ 4 ] * list[ 5 ] ) then
-      
-      return Error( "The matrix is not compatible" );
+    if NrRows( list[ 7 ] ) <> list[ 4 ] then
```
Comment (fingolfin) [inline gap/tools.gi:112]: By using NrRows, this code actually requires GAP >= 4.9 (see also <https://github.com/kamalsaleh/CddInterface/issues/11>). Of course I can avoid it, at the cost of readability. But I think requiring GAP 4.9 is reasonable -- agreed?

File: src/CddInterface.c (hunk 2 above)
Comment (kamalsaleh) [reply src/CddInterface.c:33]: I will add this suggestion as a comment to the source code. I may use it when Gap >= 4.10 is required. Currently, the package requires Gap >= 4.9 https://github.com/homalg-project/CddInterface/commit/7012a0192fa081aeaa385e13c95adc75a456bc8e.

File: gap/tools.gi (hunk 4 above)
Comment (kamalsaleh) [reply gap/tools.gi:112]: Now the package requires Gap >= 4.9. So, it adjustment is fine.

File: src/CddInterface.c [hunk 5]
```diff
@@ -444,23 +331,14 @@ static dd_MatrixPtr GapInputToMatrixPtr(Obj input)
...
-  {
-    strcpy(k_matrix, PLIST_STR(ELM_PLIST(input, 7)));
-  }
-  else
-  {
-    strcpy(k_matrix, RATPLIST_STR(ELM_PLIST(input, 7)));
-  }
-
-  strcpy(k_rowvec, RATPLIST_STR(ELM_PLIST(input, 9)));
+  strcpy(k_linearity_array, CONST_CSTR_STRING(ELM_PLIST(input, 6)));
```
Comment (kamalsaleh) [inline src/CddInterface.c]: Using **CONST_CSTR_STRING** raise warning error when compiling and tests can not be completed:
symbol lookup error: /home/kamal/Documents/local_gap_packages/pkg/CddInterface/bin/x86_64-pc-linux-gnu-default64-kv3/CddInterface.so: undefined symbol: CONST_CSTR_STRING.
I am using GAP 4.10.2 of 19-Jun-2019.
But changing CONST_CSTR_STRING to CSTR_STRING solves the problem.

**Comment by kamalsaleh** [comment]
Thank you Max. It is very nice to learn these new gap kernel vocabularies. The PR is helpfull and will be merged as soon as we clarify the use of CONST_CSTR_STRING.

File: src/CddInterface.c (hunk 5 above)
Comment (fingolfin) [reply src/CddInterface.c]: Ah, right, I forgot that `CONST_CSTR_STRING` is new in master / GAP 4.11. I'll fix the PR ASAP.

**Pull request merged**